cdc/capture/capture.go

```diff
@@ -574,6 +574,9 @@ func (c *captureImpl) resign(ctx context.Context) error {
 	failpoint.Inject("capture-resign-failed", func() {
 		failpoint.Return(errors.New("capture resign failed"))
 	})
+	if c.election == nil {
+		return nil
+	}
```

Comment:
If `resign` is caused by a processor, it will panic before this PR.
If it is caused by an owner, will it panic too?

Reply:
Yes, if c.reset() meet error between [L204 and L246](https://github.com/pingcap/tiflow/pull/7921/files#diff-4f24fb50f8768cd676f96ed676da601c83670784c3af3fac055361e7b035efd9R206-R246), capture will also panic.